scrapers/mediafusion.py

```diff
@@ -0,0 +1,231 @@
+import asyncio
+import re
+from datetime import timedelta
+from typing import List, Dict, Any
+
+import PTT
+from tenacity import RetryError
+
+from db.config import settings
+from db.models import TorrentStreams, Season, Episode, MediaFusionMetaData
+from scrapers.base_scraper import BaseScraper, ScraperError
+from utils.parser import (
+    convert_size_to_bytes,
+    is_contain_18_plus_keywords,
+)
+from utils.runtime_const import MEDIAFUSION_SEARCH_TTL
+
+
+class MediafusionScraper(BaseScraper):
+    cache_key_prefix = "mediafusion"
+
+    def __init__(self):
+        super().__init__(
+            cache_key_prefix=self.cache_key_prefix, logger_name=self.__class__.__name__
+        )
+        self.base_url = settings.mediafusion_url
+        self.semaphore = asyncio.Semaphore(10)
+
+    @BaseScraper.cache(ttl=MEDIAFUSION_SEARCH_TTL)
+    @BaseScraper.rate_limit(calls=5, period=timedelta(seconds=1))
+    async def scrape_and_parse(
```

Comment:
this need to be renamed as `_scrape_and_parse` since the `scrape_and_parse` is in parent class which will call this function